torch/csrc/autograd/variable.cpp

```diff
@@ -21,18 +21,20 @@
 
 namespace torch {
 namespace autograd {
-Variable::Impl::Impl(at::Tensor data, bool requires_grad, Edge gradient_edge)
+Variable::Impl::Impl(at::Tensor data, bool requires_grad, Edge gradient_edge, Variable::AutogradMeta* autograd_meta)
     : TensorImpl(data.type_id(), data.dtype(), /*allocator=*/nullptr, /* is variable */ true),
-      data_(std::move(data)),
-      grad_fn_(std::move(gradient_edge.function)),
-      requires_grad_(false),
-      is_view_(false),
-      output_nr_(gradient_edge.input_nr),
-      pyobj_(nullptr) {
+      data_(std::move(data)) {
+  autograd_meta->grad_fn_ = std::move(gradient_edge.function);
+  autograd_meta->requires_grad_ = false;
+  autograd_meta->is_view_ = false;
+  autograd_meta->output_nr_ = gradient_edge.input_nr;
+  autograd_meta->pyobj_ = nullptr;
+  data_.unsafeGetTensorImpl()->autograd_meta_ = autograd_meta;
+
   // set_requires_grad also checks error conditions.
   set_requires_grad(requires_grad);
   AT_CHECK(
-      !grad_fn_ || !requires_grad_,
+      !get_autograd_meta()->grad_fn_ || !get_autograd_meta()->requires_grad_,
```

Comment:
Why do you use the accessor and not the pointer that was given what to this function? Are they ever different?